### ISSUE TITLE
feat: support TypeScript 5.3.x for msw 1.3.x

### ISSUE THREAD

**Issue by SerkanSipahi** [kettanaito/msw#2025]
Loosen peer dependency constraint to unblock projects that use msw from upgrading to TypeScript 5.3.x for msw v1.3.x

**Comment by kettanaito** [comment]
Hi, @SerkanSipahi. 

We aren't going to loosen the peer dependency on TypeScript. Read more about why: https://github.com/mswjs/msw/blob/f805d349c6b9faf55c1201c51b4e7fcb29f79fe5/decisions/typescript-versioning.md

I'm marking this as a duplicate of https://github.com/mswjs/msw/issues/1875#issuecomment-1903843804. See that comment regarding updated TS support in MSW 1.x (TL;DR pull requests are welcome!).

**Comment by kettanaito** [comment]
We may reconsider the way we support TypeScript in the future. There have been a number of discussions that facilitate that this is a pain point for both developers and maintainers. Stay tuned for when that happens.

**Comment by SerkanSipahi** [comment]
> (TL;DR pull requests are welcome!).

@kettanaito see https://github.com/mswjs/msw/issues/1875#issuecomment-1935437404
